src/js/step.js

```diff
@@ -236,12 +233,11 @@ export class Step extends Evented {
    */
   show() {
     if (isFunction(this.options.beforeShowPromise)) {
-      const beforeShowPromise = this.options.beforeShowPromise();
-      if (!isUndefined(beforeShowPromise)) {
-        return beforeShowPromise.then(() => this._show());
-      }
+      return Promise.resolve(this.options.beforeShowPromise()).then(() =>
```

Comment:
Wrapping in Promise.resolve makes sure that whatever beforeshowpromise method return we can use it as a promise. If a promise is returned it's unwrapped by Promise.resolve.